### PR TITLE
Terminology Cleanup: Introduce `createAgent()` and `createOnlineAgent()` in `JenkinsRule`

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Cleans up API for JenkinsRule. I did not touch `HudsonTestCase`, because we might want to drop JUnit 3 support alltogether

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java [hunk 1]
```diff
@@ -903,39 +903,89 @@ public File createTmpDir() throws IOException {
         return env.temporaryDirectoryAllocator.allocate();
     }
 
+    // TODO replace return types once the Jenkins core is updated
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/JenkinsRule.java:906]: Do you really foresee such a class rename in core? Seems like a lot of disruption for a cosmetic issue.

**Review by jglick — COMMENTED** [review]
-0 I guess. Just adds complexity to the API surface. Would prefer to limit renames to user-facing strings.

File: src/main/java/org/jvnet/hudson/test/JenkinsRule.java (hunk 1 above)
Comment (oleg-nenashev) [reply src/main/java/org/jvnet/hudson/test/JenkinsRule.java:906]: I do not expect it anytime soon TBH. Depends on contributions though

**Comment by jglick** [comment]
More inclined to just use the new terminology incrementally as changes are made, as in #428.

**Pull request closed without merging**